### PR TITLE
Upgrade PHPStan to 0.10 and fix new violations

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Bridge/Doctrine/Orm/Extension/OrderExtension.php [hunk 1]
```diff
@@ -60,8 +60,8 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
                         $order = 'ASC';
                     }
 
-                    if (false === ($pos = \strpos($field, '.'))
-                        || isset($classMetaData->embeddedClasses[\substr($field, 0, $pos)])) {
+                    $pos = \strpos($field, '.');
```
Comment (meyerbaptiste) [inline src/Bridge/Doctrine/Orm/Extension/OrderExtension.php]: You can remove the `\`  here.

File: phpstan.neon [hunk 2]
```diff
@@ -3,16 +3,35 @@ parameters:
...
-        - '#Call to an undefined method Doctrine\\Common\\Persistence\\Mapping\\ClassMetadata::getAssociationMappings\(\)#'
+        # Real problems, hard to fix
+        - '#Parameter \#2 \$dqlPart of method Doctrine\\ORM\\QueryBuilder::add\(\) expects array\|object, string given\.#'
 
         # False positives
+        - '#Access to an undefined property Prophecy\\Prophecy\\ObjectProphecy::\$[a-zA-Z0-9_]+#'
+        - '#Access to an undefined property object::\$isIdentifierComposite.#'
         - '#Call to an undefined method Doctrine\\Common\\Persistence\\ObjectManager::getConnection\(\)#'
-        - '#Method ApiPlatform\\Core\\Bridge\\Doctrine\\Orm\\Extension\\QueryResult(Item|Collection)ExtensionInterface::supportsResult\(\) invoked with 3 parameters, 1-2 required\.#'
+        - '#Call to an undefined method PHPUnit\\Framework\\MockObject\\MockObject::[a-zA-Z0-9_]+\(\)#'
```
Comment (ondrejmirtes) [inline phpstan.neon]: What happens here? You could use phpstan-phpunit extension.

File: phpstan.neon [hunk 3]
```diff
@@ -3,16 +3,35 @@ parameters:
...
-        - '#Call to an undefined method Prophecy\\Prophecy\\ObjectProphecy::[a-zA-Z0-9_]+\(\)#'
-        - '#Access to an undefined property Prophecy\\Prophecy\\ObjectProphecy::\$[a-zA-Z0-9_]+#'
-        - '#Call to an undefined method PHPUnit\\Framework\\MockObject\\MockObject::[a-zA-Z0-9_]+\(\)#'
-        - '#Call to an undefined method Doctrine\\Common\\Persistence\\Mapping\\ClassMetadata::getAssociationMappings\(\)#'
+        # Real problems, hard to fix
+        - '#Parameter \#2 \$dqlPart of method Doctrine\\ORM\\QueryBuilder::add\(\) expects array\|object, string given\.#'
 
         # False positives
+        - '#Access to an undefined property Prophecy\\Prophecy\\ObjectProphecy::\$[a-zA-Z0-9_]+#'
+        - '#Access to an undefined property object::\$isIdentifierComposite.#'
```
Comment (ondrejmirtes) [inline phpstan.neon]: If you add property_exists, it's recognized.

Comment (dunglas) [reply phpstan.neon]: Yes but as it’s in tests/mocks, I’m
not sure it is worth it. Maybe the Prophecy extension will fix this, but it’s not compatible with 0.10 yet.

File: phpstan.neon (hunk 2 above)
Comment (dunglas) [reply phpstan.neon]: Thanks! I’ll try.

**Pull request merged**